web/modules/custom/joinup_federation/src/Plugin/pipeline/Step/AdmsValidation.php

```diff
@@ -66,23 +77,121 @@ public static function create(ContainerInterface $container, array $configuratio
   /**
    * {@inheritdoc}
    */
-  public function execute() {
+  public function initBatchProcess() {
     $graph_uri = $this->getGraphUri('sink_plus_taxo');
-    $validation = $this->admsValidator->validateByGraphUri($graph_uri);
+    $query = $this->admsValidator->getValidationQuery($graph_uri);
+    preg_match('/GRAPH.*?\{(?<where_clause>.*)\}.*?}.*?\Z/s', $query, $matches);
+    $sub_queries = explode('UNION', $matches['where_clause']);
+    $this->setBatchValue('queries', $sub_queries);
+    return ceil(count($sub_queries, self::BATCH_SIZE));
+  }
 
-    // Cleanup the 'sink_plus_taxo' graph.
-    $this->pipeline->clearGraph($this->getGraphUri('sink_plus_taxo'));
+  /**
+   * {@inheritdoc}
+   */
+  public function batchProcessIsCompleted() {
+    return !$this->getBatchValue('queries');
+  }
 
-    if ($validation->isSuccessful()) {
-      return;
+  /**
+   * {@inheritdoc}
+   */
+  public function execute() {
+    $query_array = $this->extractNextSubset('queries', self::BATCH_SIZE);
+    foreach ($query_array as $query) {
+      $query = $this->getPreparedQuery($query);
+      $graph_uri = $this->getGraphUri('sink_plus_taxo');
+      $this->admsValidator->setValidationQuery($query);
+      $validation = $this->admsValidator->validateByGraphUri($graph_uri);
```

Comment:
If we would have chaining, see AdmsValidatorInterface:: setValidationQuery() comment, this could go in 1 line.